pennylane/devices/default_tensor.py

```diff
@@ -0,0 +1,599 @@
+    _device_options = (
+        "contract",
+        "cutoff",
+        "dtype",
+        "method",
+        "max_bond_dim",
+    )
+
+    _new_API = True
+
+    def __init__(
+        self,
+        wires=None,
+        method="mps",
+        shots=None,
+        dtype=np.complex128,
+        **kwargs,
+    ) -> None:
+        if not has_quimb:
+            raise ImportError(
+                "This feature requires quimb, a library for tensor network manipulations. "
+                "It can be installed with:\n\npip install quimb"
+            )  # pragma: no cover
+
+        if not accepted_methods(method):
+            raise ValueError(
+                f"Unsupported method: {method}. The only currently supported method is mps."
+            )
+
+        if shots is not None:
+            raise ValueError("default.tensor does not support finite shots.")
+
+        if dtype not in [np.complex64, np.complex128]:
+            raise TypeError(
+                f"Unsupported type: {dtype}. Supported types are np.complex64 and np.complex128."
+            )
+
+        super().__init__(wires=wires, shots=shots)
+
+        self._method = method
+        self._dtype = dtype
+
+        # options for MPS
+        self._max_bond_dim = kwargs.get("max_bond_dim", None)
+        self._cutoff = kwargs.get("cutoff", np.finfo(self._dtype).eps)
+        self._contract = kwargs.get("contract", "auto-mps")
+
+        device_options = self._setup_execution_config().device_options
+
+        self._init_state_opts = {
+            "binary": "0" * (len(self._wires) if self._wires else 1),
+            "dtype": self._dtype.__name__,
+            "tags": [str(l) for l in self._wires.labels] if self._wires else None,
+        }
+
+        self._gate_opts = {
+            "parametrize": None,
+            "contract": device_options["contract"],
+            "cutoff": device_options["cutoff"],
+            "max_bond": device_options["max_bond_dim"],
+        }
+
+        self._expval_opts = {
+            "dtype": self._dtype.__name__,
+            "simplify_sequence": "ADCRS",
+            "simplify_atol": 0.0,
+        }
+
+        self._circuitMPS = qtn.CircuitMPS(psi0=self._initial_mps())
+
+        for arg in kwargs:
+            if arg not in self._device_options:
+                raise TypeError(
+                    f"Unexpected argument: {arg} during initialization of the default.tensor device."
+                )
+
+    @property
+    def name(self):
+        """The name of the device."""
+        return "default.tensor"
+
+    @property
+    def method(self):
+        """Supported method."""
+        return self._method
+
+    @property
+    def dtype(self):
+        """Tensor complex data type."""
+        return self._dtype
+
+    def _reset_state(self) -> None:
+        """Reset the MPS."""
+        self._circuitMPS = qtn.CircuitMPS(psi0=self._initial_mps())
+
+    def _initial_mps(self) -> qtn.MatrixProductState:
```

Comment:
```suggestion
    def _initial_mps(self) -> "qtn.MatrixProductState":
```

And similar for any other type hints.

Since I didn't have quimb installed yet, when I tried to initialize the device, I got:
```
--> 268 def _initial_mps(self) -> qtn.MatrixProductState:
    269     r"""
    270     Return an initial state to :math:`\ket{0}`.
    271 
   (...)
    275         MatrixProductState: The initial MPS of a circuit.
    276     """
    277     return qtn.MPS_computational_state(**self._init_state_opts)

NameError: name 'qtn' is not defined
```

When instead we should have the error raised during initialization.  By putting quote marks around the type hint, python doesn't actually try to import and resolve the object. It functions as a type hint, but doesn't cause dependency issues/